integration/solana/simple_collectible.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// DISCLAIMER: This file is an example of how to mint and transfer NFTs on Solana. It is not production ready and has not been audited for security.
+// Use it at your own risk.
+
+import '../../solana-library/spl_token.sol';
+
+contract SimpleCollectible {
+    // On Solana, an associated token account saves the amount of tokens an owner has.
+    // It also holds the mint account and the owner account public keys.
+    // The associated token account and the mint account save all the information necessary to prove the authenticty of an NFT.
+    address private associatedTokenAccount;
+    // A resource identifier to access the NFT. It could be any other data to be saved on the blockchain
+    string private uri;
+
+    // These events log on the blockchain transactions made with this NFT
+    event NFTMinted(address owner, address mintAccount);
+    event NFTSold(address from, address to);
+
+    // The associated token account will identify the NFT in this example
+    constructor (address _associatedTokenAccount) {
+        associatedTokenAccount = _associatedTokenAccount;
+    }
+
+    /// Create a new NFT and associate it to an URI
+    ///
+    /// @param tokenURI a URI that leads to the NFT resource
+    /// @param mintAuthority an account that signs each new mint
+    function createCollectible(string memory tokenURI, address mintAuthority) public {
+        SplToken.AssociatedAccountData data = SplToken.get_associated_account_data(associatedTokenAccount);
+
+        // An NFT on Solana is a SPL-Token with only one minted token.
+        // The mintAccount saves information about the token (e.g. quantity of tokens avaialble)
+        SplToken.mint_to(data.mintAccount, associatedTokenAccount, mintAuthority, 1);
```

Comment:
In order to be sure it's non-fungible, you may want to check that the supply of the mint is currently 0 before doing the mint